src/screens/Technology/components/Rating/index.tsx

```diff
@@ -1,94 +1,431 @@
+                <S.NegativeText>Pontos negativos</S.NegativeText>
+                {review.negative.map((comment: string, idx: number) => (
+                  <S.PositiveNegativeWrapper key={`review_${review.id}_n${idx}`}>
+                    <S.PositiveNegativeText>{comment}</S.PositiveNegativeText>
+                  </S.PositiveNegativeWrapper>
+                ))}
+              </View>
+            ))}
+          </>
+        ) : (<><S.RateCommentText>Nenhum comentário encontrado.</S.RateCommentText></>)}
+      </S.ReviewsWrapper>
+    </>
+  );
+};
+
+const Rating = (): JSX.Element => {
+  const technology = useTechnology();
+  const [reviews, setReviews] = useState(null);
+  const [rating, setRating] = useState(null);
+
+  const [showModal, setShowModal] = useState<boolean>(false);
+
+  const calcRating = (data: ReviewProps[]) => {
+    if (data && data.length) setRating(data.reduce((acc, cur) => acc + cur.rating, 0.0) / 2);
+  };
+
+  const loadReviews = useCallback(
+    async () => {
+      await getReviews(technology.id).then((data) => {
+        setReviews(data);
+        calcRating(data);
+      });
+    },
+    [technology, reviews, rating],
+  );
+
+  useEffect(() => {
+    loadReviews();
+  }, [technology, rating]);
+
+  return (
+    <S.Wrapper>
+      <S.Title>Avaliações</S.Title>
+      {reviews && reviews.length ? (
+        <>
+          <S.WrapperRate>
+            <S.RateNumber>{rating}</S.RateNumber>
+            <S.RateStars>
+              <Stars initialScore={rating} size={24} />
+              <S.RateStarsDescription>
+                Média entre
+                {' '}
+                {reviews && reviews.length}
+                {' '}
+                {reviews && reviews.length > 1 ? 'opiniões' : 'opinião'}
+              </S.RateStarsDescription>
+            </S.RateStars>
+          </S.WrapperRate>
+          <Reviews data={reviews} />
+          <S.RateViewMore activeOpacity={0.7}>
+            <S.RateViewMoreText>Ver todas as opiniões</S.RateViewMoreText>
+          </S.RateViewMore>
+        </>
+      ) : (
+        <View style={{ paddingHorizontal: 16, marginBottom: 10 }}><S.RateCommentText>Nenhuma avaliação encontrada.</S.RateCommentText></View>
+      )}
+
+      <S.AnswersViewMore activeOpacity={0.7} onPress={() => setShowModal(true)}>
+        <Ionicons name="star-outline" size={20} color={Colors.primary} />
+        <S.AnswersViewMoreText>
+          Deixe a sua opinião
+        </S.AnswersViewMoreText>
+      </S.AnswersViewMore>
+
+      <Modal
+        title="Deixe sua avaliação"
+        titleStyle={{
+          width: 200,
+          fontSize: 20,
+          lineHeight: 26,
+        }}
+        height={800}
+        animationType="slide"
+        visible={showModal}
+        onClose={() => { setShowModal(false); loadReviews(); }}
+      >
+        <S.ModalContent>
+          <ReviewForm onFinish={setShowModal} />
+        </S.ModalContent>
+      </Modal>
+
+    </S.Wrapper>
+  );
+};
```

Comment:
This file is giant! It would be interesting to refactor it 🤔

Reply:
Good idea. I'll do soon. Let's clean up the issues.